src/test/resources/http-listener-message-properties-config.xml

```diff
@@ -14,6 +14,9 @@
     <http:listener-config name="listenerBaseConfig" basePath="api">
         <http:listener-connection host="localhost" port="${port2}"/>
     </http:listener-config>
+    <http:listener-config name="listenerEncodedBaseConfig" basePath="a p i">
```

Comment:
this will be matched only for request that have this part properly encoded, right?

Reply:
Yes, the test is hitting “a%20p%20i”

Reply:
I added this test after realizing that listener paths are stored as defined (decoded if you will) so if we try to extract the base path from an encoded request path it wouldn’t work.